pkg/action/archive.go

```diff
@@ -0,0 +1,225 @@
+package action
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/ulikunitz/xz"
+)
+
+const maxBytes = 1 << 29 // 512MB
+
+// copyArchive copies the source archive file to the temporary directory.
+func copyArchive(src string, dst string) error {
+	r, err := os.Open(src)
+	if err != nil {
+		return fmt.Errorf("failed to open source file: %w", err)
+	}
+	defer r.Close()
+
+	w, err := os.CreateTemp(dst, fmt.Sprintf("%s.*", filepath.Base(src)))
+	if err != nil {
+		return fmt.Errorf("failed to create temporary file: %w", err)
+	}
+
+	defer func() {
+		if cerr := w.Close(); cerr != nil {
+			log.Printf("failed to close file: %v", cerr)
+		}
+	}()
+
+	if _, err = io.Copy(w, r); err != nil {
+		return fmt.Errorf("failed to copy data: %w", err)
+	}
+
+	return nil
+}
+
+// tempDir creates a temporary directory.
```

Comment:
This comment is a bit misleading since it not only creates the directory, but also copies the archive to it.

Reply:
Updated in [`424a907` (#144)](https://github.com/chainguard-dev/bincapz/pull/144/commits/424a907f4bbc7201f6f4bc242c2e2b0ac06ec853).